Mathlib/Combinatorics/SimpleGraph/Basic.lean

```diff
@@ -539,6 +539,13 @@ theorem edgeSet_bot : (⊥ : SimpleGraph V).edgeSet = ∅ :=
   Sym2.fromRel_bot
 #align simple_graph.edge_set_bot SimpleGraph.edgeSet_bot
 
+@[simp]
+theorem edgeSet_top : (⊤ : SimpleGraph V).edgeSet = {e | ¬e.IsDiag} := by
+  ext f
+  constructor <;> intro hm
+  · simp [(⊤ : SimpleGraph V).not_isDiag_of_mem_edgeSet hm]
+  · obtain ⟨_, _⟩ := f; exact hm
```

Comment:
If you do induction on `f`, aesop should be able to handle it.

Reply:
This is clearly impossible. `f`'s type is `Sym2 V` which is not an inductive type.

Reply:
Do you know about [`Quotient.inductionOn`](https://leanprover-community.github.io/mathlib4_docs/Init/Core.html#Quotient.inductionOn)?

Reply:
Yeah, now everything works.